src/opt/ReplaceHalfMaskedMemOps.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "ReplaceHalfMaskedMemOps.h"
+
+#include <llvm/IR/IRBuilder.h>
+
+namespace ispc {
+
+bool lIsSecondHalfAllFalse(llvm::Value *mask) {
```

Comment:
The function's name suggests it only checks if the second half of the vector is zero, but the code also checks if the first half contains all ones. Either adjust the name and documentation to reflect this dual purpose or separate the functionality into two distinct functions, depending on your use case.

Reply:
I have changed that in the last revision completely. I have realized that we need to support more cases, e.g., when only 4 first elements are true. It is needed to generate code using narrow registers for targets with `x16`.